Fika.Core/Coop/GameMode/CoopGame.cs

```diff
@@ -203,22 +207,141 @@ public async Task CreateCoopHandler()
 
         public Dictionary<string, Player> Bots { get; set; } = [];
 
+        private List<CoopPlayer> GetPlayers(CoopHandler coopHandler)
+        {
+            List<CoopPlayer> humanPlayers = new List<CoopPlayer>();
+
+            // Grab all players
+            foreach (CoopPlayer player in coopHandler.Players.Values)
+            {
+                if ((player.IsYourPlayer || player is ObservedCoopPlayer) && player.HealthController.IsAlive)
+                {
+                    humanPlayers.Add(player);
+                }
+            }
+            return humanPlayers;
+        }
+
+        private float GetDistanceFromPlayers(Vector3 position, List<CoopPlayer> humanPlayers)
+        {
+            float distance = float.PositiveInfinity;
+
+            foreach (Player player in humanPlayers)
+            {
+                float tempDistance = Vector3.SqrMagnitude(position - player.Position);
+
+                if (tempDistance < distance) // Get the closest distance to any player. so we dont despawn bots in a players face.
+                {
+                    distance = tempDistance;
+                }
+            }
+            return distance;
+        }
+
+        private string GetFurthestBot(Dictionary<string, Player> bots, CoopHandler coopHandler, out float furthestDistance)
+        {
+            string furthestBot = string.Empty;
+            furthestDistance = 0f;
+
+            List<CoopPlayer> humanPlayers = GetPlayers(coopHandler);
+
+            foreach (var kvp in Bots)
```

Comment:
Are we using the key value outside of line 260?
Might as well only refer to `Bots.Values` and use the `ProfileId` property. 

If not, change `kvp` into something more coherent, e.g. `botsDict`.

Reply:
changed the name to ``botKeyValuePair``